cedar-policy-core/src/ast/policy.rs

```diff
@@ -1694,7 +1702,10 @@ impl EntityReference {
     pub fn into_expr(&self, slot: SlotId) -> Expr {
         match self {
             EntityReference::EUID(euid) => Expr::val(euid.clone()),
-            EntityReference::Slot => Expr::slot(slot),
+            EntityReference::Slot(loc) => loc
+                .as_ref()
+                .map(|loc| ExprBuilder::new().with_source_loc(loc.clone()).slot(slot))
+                .unwrap_or(Expr::slot(slot)),
```

Comment:
maybe could be simplified to something like
```suggestion
            EntityReference::Slot(loc) => Expr::slot(slot).with_maybe_source_loc(loc.cloned()),
```